core/Date.php

```diff
@@ -1070,7 +1070,12 @@ private static function getInvalidDateFormatException($dateString)
         return new Exception($message . ": $dateString");
     }
 
-    private static function getNowTimestamp()
+    /**
+     * For tests.
+     * @return int|null
+     * @ignore
+     */
+    public static function getNowTimestamp()
     {
         return isset(self::$now) ? self::$now : time();
```

Comment:
In `updateStartTime` we use `Date::now()->getTimestampUTC();`. shouldn't we make sure to use the same logic? Likely it won't be an issue and in the end it is maybe the same logic though.